src/proposals/20241114/SparkEthereum_20241114.t.sol

```diff
@@ -0,0 +1,428 @@
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.ALM_PROXY), 0);
+        assertEq(IERC20(Base.USDS).balanceOf(Base.PSM3),       USDS_BRIDGE_AMOUNT);
+        assertEq(IERC20(Base.SUSDS).balanceOf(Base.PSM3),      susdsShares);
+
+        // Ensure that value controlled by the ALM proxy in the PSM is equal to total value deposited
+        uint256 proxyPsmShares = IPSMLike(Base.PSM3).shares(Base.ALM_PROXY);
+        assertEq(proxyPsmShares, susdsPsmShares + 1_000_000e18);
+        assertEq(
+            IPSMLike(Base.PSM3).convertToAssetValue(proxyPsmShares),
+            mainnetSUsdsAssets + 1_000_000e18
+        );
+    }
+
+    function testDepositUSDCPSM3() public {
+        _setupCrossChainTest();
+
+        mainnet.selectFork();
+        uint256 susdsShares = IERC4626(Ethereum.SUSDS).convertToShares(SUSDS_DEPOSIT_AMOUNT);
+
+        // Do initial deposit of USDS and sUSDS into PSM3
+        base.selectFork();
+
+        OptimismBridgeTesting.relayMessagesToDestination(nativeBridge, true);
+
+        vm.startPrank(RELAYER);
+        ForeignController(Base.ALM_CONTROLLER).depositPSM(Base.USDS,  USDS_BRIDGE_AMOUNT);
+        ForeignController(Base.ALM_CONTROLLER).depositPSM(Base.SUSDS, susdsShares);
+        vm.stopPrank();
+
+        mainnet.selectFork();
+
+        // Mint and bridge USDC
+        uint256 usdcAmount = 800_000e6;
+        uint256 usdcSeed   = 1e6;
+
+        vm.startPrank(RELAYER);
+        MainnetController(Ethereum.ALM_CONTROLLER).mintUSDS(usdcAmount * 1e12);
+        MainnetController(Ethereum.ALM_CONTROLLER).swapUSDSToUSDC(usdcAmount);
+        MainnetController(Ethereum.ALM_CONTROLLER).transferUSDCToCCTP(usdcAmount, CCTPForwarder.DOMAIN_ID_CIRCLE_BASE);
+        vm.stopPrank();
+
+        base.selectFork();
+
+        assertEq(IERC20(Base.USDC).balanceOf(Base.ALM_PROXY), 0);
+        assertEq(IERC20(Base.USDC).balanceOf(Base.PSM3),      usdcSeed);
+
+        // FIXME this is causing a MemoryOOG error, doing a workaround for now
+        //CCTPBridgeTesting.relayMessagesToDestination(cctpBridge, true);
+        vm.prank(0xF977814e90dA44bFA03b6295A0616a897441aceC);  // Some USDC whale on Base
+        IERC20(Base.USDC).transfer(Base.ALM_PROXY, usdcAmount);
+
+        uint256 proxyAssets = IPSMLike(Base.PSM3).convertToAssetValue(IPSMLike(Base.PSM3).shares(Base.ALM_PROXY));
+
+        assertEq(IERC20(Base.USDC).balanceOf(Base.ALM_PROXY), usdcAmount);
+        assertEq(IERC20(Base.USDC).balanceOf(Base.PSM3),      usdcSeed);
+
+        vm.startPrank(RELAYER);
+        ForeignController(Base.ALM_CONTROLLER).depositPSM(Base.USDC, usdcAmount);
+        vm.stopPrank();
+
+        assertEq(IERC20(Base.USDC).balanceOf(Base.ALM_PROXY), 0);
+        assertEq(IERC20(Base.USDC).balanceOf(Base.PSM3),      usdcSeed + usdcAmount);
+
+        uint256 proxyAssetsAfter = IPSMLike(Base.PSM3).convertToAssetValue(IPSMLike(Base.PSM3).shares(Base.ALM_PROXY));
+        assertEq(proxyAssetsAfter, proxyAssets + usdcAmount * 1e12);
+    }
+
```

Comment:
We should add a test here that bridges funds from base to mainnet to demonstrate that mintRecipient in that direction is configured correctly

Reply:
Good idea.